MatrixSDK/MXSession.m

```diff
@@ -2217,6 +2224,21 @@ - (MXHTTPOperation*)joinRoom:(NSString*)roomIdOrAlias
                      success:(void (^)(MXRoom *room))success
                      failure:(void (^)(NSError *error))failure {
     
+    if ([self isJoinedOnRoom:roomIdOrAlias])
```

Comment:
Why not using `RoomSummary.membershipTransitionState` and `MXMembershipTransitionStateJoining`?

Reply:
Updated for both is joined and is invited logic.